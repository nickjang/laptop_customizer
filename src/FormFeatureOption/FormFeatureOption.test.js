import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FormFeatureOption from './FormFeatureOption';

describe('<FormFeatureOption />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<FormFeatureOption />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders empty when no props supplied', () => {
    const wrapper = shallow(
    <FormFeatureOption />
      );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders option when props are supplied', () => {
    const wrapper = shallow(
    <FormFeatureOption key={1} itemHash="feature-1" feature="Display" item={{name:'15', cost:10}}
      selectedFeatureName={'10'}/>
      );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});