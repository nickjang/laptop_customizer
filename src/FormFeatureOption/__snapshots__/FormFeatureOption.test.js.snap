// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormFeatureOption /> renders empty when no props supplied 1`] = `
<div
  className="feature__item"
>
  <input
    checked={true}
    className="feature__option"
    id=""
    name=""
    onChange={[Function]}
    type="radio"
  />
  <label
    className="feature__label"
    htmlFor=""
  >
     (
    $0.00
    )
  </label>
</div>
`;

exports[`<FormFeatureOption /> renders option when props are supplied 1`] = `
<div
  className="feature__item"
>
  <input
    checked={false}
    className="feature__option"
    id="feature-1"
    name="Display"
    onChange={[Function]}
    type="radio"
  />
  <label
    className="feature__label"
    htmlFor="feature-1"
  >
    15
     (
    $10.00
    )
  </label>
</div>
`;
